x-pack/plugins/fleet/server/services/agent_policy.ts

```diff
@@ -1070,6 +1080,104 @@ class AgentPolicyService {
     };
   }
 
+  public async createAgentlessAgent(
+    esClient: ElasticsearchClient,
+    soClient: SavedObjectsClientContract,
+    agentlessAgentPolicy: AgentPolicy
+  ) {
+    if (!appContextService.getCloud()?.isCloudEnabled) {
```

Comment:
please add log messages of level debug, info, warn so we could easily trace the logic here and troubleshoot failures going forward.

Specially I recommend to log before and after the request with the response payload as well